docs/hello_nextflow/06_hello_config.md

```diff
@@ -0,0 +1,266 @@
+
+### 3.1.1 Set process resources
+
+By default, Nextflow will use a single CPU and 2GB of memory for each process. We can modify this behaviour by setting the `cpu` and `memory` directives in the `process` block. Add the following to the end of your `nextflow.config` file:
+
+```groovy title="nextflow.config" linenums="11"
+process {
+    cpus = 8
+    memory = 4GB
+}
+```
+
+Run the pipeline again with the modified configuration:
+
+```bash
+nextflow run seqeralabs/nf-hello-gatk -profile docker
+```
+
+You shouldn't see any difference, however you might notice the three processes get bottlenecked behind each other. This is because Nextflow will make sure we aren't using more than CPUs than are available.
+
+### 3.1.2 Modify process resources for a specific process
+
+We can also modify the resources for a specific process using the `withName` directive. Add the following to the end of your `nextflow.config` file:
+
+```groovy title="nextflow.config" linenums="11"
+process {
+    withName: 'GATK_HAPLOTYPECALLER' {
+        cpus = 8
+        memory = 4.GB
+    }
+}
+```
+
+Run the pipeline again with the modified configuration:
+
+```bash
+nextflow run seqeralabs/nf-hello-gatk -profile docker
+```
+
+Now the settings are only applied to the GATK HaplotypeCaller process. This is useful when your processes have different resource requirements, so you can right-size your resources for each process.
+
+## 4. Executor
+
+### 4.1. Local executor
+
+Up until now, we have been running our pipeline with the local executor. This runs each step on the same machine that Nextflow is running on. However, for large genomics pipelines, you will want to use a distributed executor. Nextflow supports a number of different distributed executors, including:
+
+-   HPC (SLURM, PBS, SGE)
+-   AWS Batch
+-   Google Batch
+-   Azure Batch
+-   Kubernetes
+
+We can modify the executor used by nextflow using the `executor` process directive. Until now, we have been using the `local` executor (default). The following configuration is implied:
+
+```groovy title="nextflow.config" linenums="18"
+process {
+    executor = 'local'
+}
+```
+
+### 4.2. Other executors
+
+!!! note
+
+    This is a demonstration and designed to go wrong!
+
+If we wish to change executor, we could simply set this to one of the values in the documentation:
+
+```groovy title="nextflow.config" linenums="18"
+process {
+    executor = 'slurm'
+}
+```
+
+However, if we add this to our config and run the pipeline we will that includes this error:
+
+```console
+Cannot run program "sbatch"
+```
+
+Nextflow has interpreted that we wish to submit to a Slurm cluster, which requires te use of the command `sbatch`. However, because we are not _actually_ using Slurm the `sbatch` command is not installed therefore there is an error.
```

Comment:
```suggestion
Nextflow has interpreted that we wish to submit to a Slurm cluster, which requires te use of the command `sbatch`. 
However, because our Gitpod instance doesn't have slurm installed (and isn't connected to a cluster)  this throws an error.
```